src/publish.h

```diff
@@ -0,0 +1,35 @@
+#ifndef PUBLISH_H
+#define PUBLISH_H
+
+#ifndef ZMQ
+#define report(type,msg) ;
+
+#else
+#define report(type,msg) g->remote.send(type,msg);
+
+#include <string>
+
+class publish
+{
+    public:
+        publish() = default;
+        ~publish();
+
+        publish( publish && );
+        publish &operator=( publish && );
+
+        publish( const publish & ) = delete;
+        publish &operator= ( const publish & ) = delete;
+
+        explicit publish( const std::string &remote );
+
+        void send( const std::string &type, const std::string &msg );
+
+    private:
+        void *ctx = nullptr; /** ZeroMQ context */
```

Comment:
These pointers should be in `unique_ptr` with a proper deleter function. See `WINDOW_PTR` for an example. That would give you automatic move assignment/constructor and also automatically forbid copy assignment/constructor. And it would make a much shorter constructor. And an automatic destructor.